DEHPEcosimPro/ViewModel/Rows/ParameterDifferenceViewModel.cs

```diff
@@ -0,0 +1,324 @@
+
+    using DevExpress.Mvvm.Native;
+
+    /// <summary>
+    /// Object ot use in MainWindow, Value DiffS
+    /// </summary>
+    public class ParameterDifferenceViewModel
+    {
+
+        /// <summary>
+        /// The <see cref="IDstController"/>
+        /// </summary>
+        private readonly IDstController dstController;
+
+        /// <summary>
+        /// The Thing already on the data hub
+        /// </summary>
+        private Parameter oldThing;
+
+        /// <summary>
+        /// The Thing already on the data hub
+        /// </summary>
+        public Parameter OldThing
+        {
+            get => this.oldThing;
+            set => this.oldThing = value;
+        }
+
+        /// <summary>
+        /// The thing from Ecosimpro
+        /// </summary>
+        private Parameter newThing;
+
+        /// <summary>
+        /// The thing from Ecosimpro
+        /// </summary>
+        public Parameter NewThing
+        {
+            get => this.newThing;
+            set => this.newThing = value;
+        }
+
+        /// <summary>
+        /// List of value from <see cref="NewThing"/>, dependant of states and options
+        /// </summary>
+        private readonly List<IValueSet> listofsetOfNewValues = new List<IValueSet>();
+
+        /// <summary>
+        /// List of value from <see cref="OldThing"/>, dependant of states and options
+        /// </summary>
+        private readonly List<IValueSet> listofsetOfOldValues = new List<IValueSet>();
+
+        private List<ParameterDifferenceRowViewModel> listOfParameters = new List<ParameterDifferenceRowViewModel>();
+
+        /// <summary>
+        /// List of <see cref="ParameterDifferenceRowViewModel"/> to show in MainWindow,
+        /// multiple item have the same Iid because the set of data can be different due to states and options
+        /// </summary>
+        public List<ParameterDifferenceRowViewModel> ListOfParameters
+        {
+            get => this.listOfParameters;
+            set => this.listOfParameters = value;
+        }
+
+        /// <summary>
+        /// Evaluate if Things have Options or States, and compute data in List of <see cref="ParameterDifferenceRowViewModel"/>
+        /// </summary>
+        /// <param name="OldThing"></param>
+        /// <param name="NewThing"></param>
+        /// <param name="dstController"></param>
+        public ParameterDifferenceViewModel(Parameter OldThing, Parameter NewThing, IDstController dstController)
+        {
+            this.OldThing = OldThing;
+            this.NewThing = NewThing;
+            this.dstController = dstController;
+
+            this.InitializeOptionAndStateDependency();
+        }
+
+        /// <summary>
+        /// Determine if Thing is Option and/or State Dependent and construct list of parameters accordingly
+        /// </summary>
+        private void InitializeOptionAndStateDependency()
+        {
+            var isoptiondependant = this.NewThing.IsOptionDependent;
```

Comment:
what happens if `NewThing` is null?

Reply:
you're right, but i'll handle it in DifferenceViewModel